osrf/vrx
# [Garden] Port AcousticPinger plugin

This pull request ports the `AcousticPingerPlugin` to Gazebo.

### How to test it?

1. Run the simulation loading the new `gymkhana_task` world:
```
ros2 launch vrx_gz competition.launch.py world:=gymkhana_task
```

2. On a separate terminal echo the `pinger` topic:
```
ros2 topic echo /wamv/pingers/pinger/range_bearing
```

The pinger is located 2 meters below the red buoy. You should get a range of around 29m. You can also try rotating/translating the WAM-V to see how the angles change. Keep in mind that there's some noise applied.

3. You can also try changing the pinger position (for example to the origin):
```
ros2 topic pub /wamv/pingers/pinger/set_pinger_position --once geometry_msgs/msg/Vector3
```

4. Verify how the range changes accordingly.
